lightwood/api/predictor.py

```diff
@@ -362,7 +358,7 @@ def type_map(col_name):
 
                     # Once we are past the priming/warmup period, start training the selfaware network
 
-                    if subset_iteration > 1 and not mixer.is_selfaware and CONFIG.SELFAWARE and not mixer.stop_selfaware_training and training_error < 0.35:
+                    if subset_iteration > 1 and not mixer.is_selfaware and self.config['mixer']['selfaware'] and not mixer.stop_selfaware_training and training_error < 0.35:
```

Comment:
I think the idea is that the predictor API, does not know about self-aware at all, so we should abstract this a little better, we can talk about it in the call, we can make this change in a later PR

Reply:
I mean... I agree that the whole loop in Predictor.py was ugly... I wanted to get rid of it and do it similarly to how I do it in the trainable-encoders (using the gym.py implementation).

But it ends up being complicated so until we have a second "main" mixer I don't think it's worth bothering, or at least until we get time to refactor. So I agree, not ideal, but also agree that we should leave it for another PR.